psi4/src/psi4/libmints/matrix.cc

```diff
@@ -3403,8 +3403,49 @@ SharedMatrix doublet(const SharedMatrix &A, const SharedMatrix &B, bool transA,
 
 SharedMatrix triplet(const SharedMatrix &A, const SharedMatrix &B, const SharedMatrix &C, bool transA, bool transB,
                      bool transC) {
-    SharedMatrix T = doublet(A, B, transA, transB);
-    SharedMatrix S = doublet(T, C, false, transC);
+
+    int dim1 = A->nrow();
+    int dim2 = A->ncol();
+    int dim3 = B->nrow();
+    int dim4 = B->ncol();
+    int dim5 = C->nrow();
+    int dim6 = C->ncol();
+
+    int temp = 0;
+
+    if (transA) {
+        temp = dim1;
+        dim1 = dim2;
+        dim2 = temp;
+    }
+
+    if (transB) {
+        temp = dim3;
+        dim3 = dim4;
+        dim4 = temp;
+    }
+
+    if (transC) {
+        temp = dim5;
+        dim5 = dim6;
+        dim6 = temp;
+    }
```

Comment:
Here you've flipped everything so that the operation is (1 x 2) x (3 x 4) = (5 x 6).

Reply:
... scratch that,  (1 x 2) x (3 x 4) x (5 x 6).